src/divert.c

```diff
@@ -8,7 +8,7 @@
 #define MAX_PACKETSIZE 0xFFFF
 #define READ_TIME_PER_STEP 3
 // FIXME does this need to be larger then the time to process the list?
-#define CLOCK_WAITMS 40
+#define CLOCK_WAITMS 1
```

Comment:
I believe this value is too low. For example this will trigger excessive logging due to the condition on line 232 being always true. It'll also lead to elevated CPU utilization since the timer thread will essentially be busy looping most of the time. The only reason this won't cause real issues is because the Windows scheduler does not obey it and can easily sleep for 10ms or more at a time. As a practical suggestion, I'd recommend a value of 10ms instead.

Reply:
Hello.

I ran some ping tests to compare the accuracy of simulated delays between a CLOCK_WAITMS value of 10 ms and 1 ms.

On my test system, with the 10 ms value, the simulated delays are accurate within about 6 ms.

With the 1 ms value, the accuracy increases to about 2 ms.

I initially shared your concerns about increased CPU usage, but my testing so far has shown that it seems OK.

Using the 1 ms CLOCK_WAITMS value, I compared the CPU load in a session with and without Clumsy by monitoring the task manager CPU performance graph. I could not see any notable CPU increase or decrease as I started and stopped Clumsy.

Of course, things may behave differently on different systems, so the steps to test for this Jira ticket will include SQA comparing old and new Clumsy on various different computers, to be sure the accuracy is increased, and the CPU is not increased very much.

Does that sound good?

Reply:
@agnickolov You've already approved the PR, but I'll hold off on merging until you have a chance to respond to my comment above. Thanks.

Reply:
Sure, go ahead.